### PR TITLE
added ne_50m_physical_geography_marine_polys

### PR DESCRIPTION
**Pull request by jbusecke**
Adds ocean regions to the natural earth module.

### CODE REVIEW

**Comment by jbusecke** [comment]
I also noticed that there are marine polys for 110m and 10m available. Should I add those aswell?

File: docs/defined_ocean_basins.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+################################
+Marine Areas/Ocean Basins (NaturalEarth)
+################################
```
Comment (mathause) [inline docs/defined_ocean_basins.rst]:
```suggestion
########################################
Marine Areas/Ocean Basins (NaturalEarth)
########################################
```

I think they need the same length as the title.

**Comment by jbusecke** [comment]
I can confirm that they are different polygons! So they should not be removed. Either they should be renamed (e.g. West/East Ross Sea) or joined to a single polygon. Do you think it would be worth raising an issue upstream?

File: docs/defined_ocean_basins.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    mask = states.mask(lon, lat, wrap_lon=True)
+    mask_ma = np.ma.masked_invalid(mask)
+
+    basins.plot(add_label=False, add_ocean=False);
+
+    LON_EDGE = np.arange(0, 360)
+    LAT_EDGE = np.arange(90, -90, -1)
+
+    plt.pcolormesh(LON_EDGE, LAT_EDGE, mask_ma, cmap='viridis');
```
[the remark this answers is not in the document]
Comment (jbusecke) [reply docs/defined_ocean_basins.rst]: Ah nice.

**Comment by jbusecke** [comment]
I added the 110m, but noticed it has quite a lot of gaps, so I refrained from using it in the example. I think the 50m version is the much better choice. Also added the doc page to the index and updated tests.

File: regionmask/defined_regions/natural_earth.py [hunk 3]
```diff
@@ -213,5 +216,37 @@ def land_110(self):
...
+            opt = dict(
+                resolution="50m",
+                category="physical",
+                name="geography_marine_polys",
+                title="Natural Earth: ocean basins 50m",
+                names="name",
+                abbrevs="name",
+            )
+
+            self._ocean_basins_50 = _obtain_ne(**opt)
```
Comment (mathause) [inline regionmask/defined_regions/natural_earth.py]: For the Mediterranean I would do it like so:

```suggestion
            regs = _obtain_ne(**opt)
            
            # NOTE: naturalearth includes duplicate names

            # raise an error if naturalearth changes the name
            msg = "naturalearth renamed this region, please raise an issue in regionmask"
            assert regs[14].name == 'Mediterranean Sea' , msg

            regs[14].name = 'Mediterranean Sea East'                                                                                                                                                    
            regs[14].abbrev = 'Mediterranean Sea East'
            # dito for regs[30] 
            self._ocean_basins_50 = regs
```

File: docs/defined_ocean_basins.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
+########################################
+Marine Areas/Ocean Basins (NaturalEarth)
+########################################
+
+The outline of the marine areas are obtained from
+`Natural Earth <http://www.naturalearthdata.com/>`_.
+They are automatically downloaded on-the-fly (but only once) with cartopy and opened with geopandas.
+The following marine regions are defined in regionmask:
+
+* Marine Areas 1:50m
```
Comment (mathause) [inline docs/defined_ocean_basins.rst]:
```suggestion
* Marine Areas 1:50m
* Marine Areas 1:110m
```

File: regionmask/defined_regions/natural_earth.py (hunk 3 above)
Comment (jbusecke) [reply regionmask/defined_regions/natural_earth.py]: This seems like a great workaround until they fix it upstream.

Comment (jbusecke) [reply regionmask/defined_regions/natural_earth.py]: For the time beeing. Should we just do the same thing with the Ross Sea? Ill raise an issue upstream.

Comment (jbusecke) [reply regionmask/defined_regions/natural_earth.py]: Upstream issue here: https://github.com/nvkelso/natural-earth-vector/issues/319

**Comment by jbusecke** [comment]
I have removed the 110 again, added renaming for both Ross and Med including the warning. 
Applying the merging dict yields this:
image
This is pretty much exactly what I needed! Thanks a lot for the help.

File: regionmask/defined_regions/natural_earth.py [hunk 5]
```diff
@@ -213,5 +215,44 @@ def land_110(self):
...
+            # dito for regs[30]
+            regs[30].name = "Mediterranean Sea Western Basin"
+            regs[30].abbrev = "Mediterranean Sea Western Basin"
+
+            assert regs[26].name == "Ross Sea", msg
+            regs[26].name = "Ross Sea Eastern Basin"
+            regs[26].abbrev = "Ross Sea Eastern Basin"
+            # dito for regs[30]
+            regs[29].name = "Ross Sea Western Basin"
+            regs[29].abbrev = "Ross Sea Western Basin"
```
Comment (mathause) [inline regionmask/defined_regions/natural_earth.py]:
```suggestion

           # rename the "Mediterranean Sea" region
            assert regs[14].name == "Mediterranean Sea", msg
            regs[14].name = "Mediterranean Sea Eastern Basin"
            regs[14].abbrev = "Mediterranean Sea Eastern Basin"

            assert regs[30].name == "Mediterranean Sea", msg
            regs[30].name = "Mediterranean Sea Western Basin"
            regs[30].abbrev = "Mediterranean Sea Western Basin"

           # rename the "Ross Sea" region
            assert regs[26].name == "Ross Sea", msg
            regs[26].name = "Ross Sea Eastern Basin"
            regs[26].abbrev = "Ross Sea Eastern Basin"
            
            assert regs[29].name == "Ross Sea", msg
            regs[29].name = "Ross Sea Western Basin"
            regs[29].abbrev = "Ross Sea Western Basin"
```

I'm a bit paranoid renaming something that we shouldn't so I'd like to have checks for all of them

Comment (jbusecke) [reply regionmask/defined_regions/natural_earth.py]: Oh yeah, even better.

**Comment by jbusecke** [comment]
@mathause , do you think this is good to go? Or do you think we need some more changes? I also wanted to thank you for all the help implementing this!

**Comment by mathause** [comment]
Looks good to me! Thanks a lot for your contribution @jbusecke

**Comment by jbusecke** [comment]
Seems like the docs are not built yet. Is there a trigger for it?

**Comment by jbusecke** [comment]
Ah it seems like it fails due to high memory consumption. Do you think this is due to changes I made?

**Comment by mathause** [comment]
Thanks for the reminder & no, that's not your fault. I feel I spend more time getting readthedocs to run than working on the package... :( I'll have to ask for more ressources, I already reduced the usage of conda to the minimum I could - this did work 6 weeks ago...

**Comment by jbusecke** [comment]
Ugh sorry to hear that...Let me know if I can do something.

**Comment by mathause** [comment]
I got more resources and it works again - nice. I already fixed the white line and removed the colorbar, but I'll render that later.

**Pull request merged**